Build failure: nvidia-open

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

- Use my [nix-config](github.com/NikoPit/nix-config)
- do `sudo nix flake update`
- rebuild nixos with my nix config

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/321915381

### Relevant log output

```console
error: Cannot build '/nix/store/5skga05vis5mfv1p8mqm98hikbw1rfg3-nvidia-open-6.19.2-580.126.09.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/jr589g6l8x8ccr2f9kn4la1w4pipvs7j-nvidia-open-6.19.2-580.126.09
       Last 25 log lines:
       > nvidia-uvm/uvm_hmm.c:2159:9: note: in expansion of macro 'ZONE_DEVICE_PAGE_INIT'
       >  2159 |         ZONE_DEVICE_PAGE_INIT(dpage);
       >       |         ^~~~~~~~~~~~~~~~~~~~~
       > In file included from /nix/store/jm1r8xwcalll7b85h9gippv50w7ixwzb-linux-6.19.2-dev/lib/modules/6.19.2/source/include/linux/mm.h:33,
       >                  from ././common/inc/nv-pgprot.h:30,
       >                  from ././common/inc/nv-linux.h:33,
       >                  from nvidia-uvm/uvm_linux.h:40,
       >                  from nvidia-uvm/uvm_common.h:43,
       >                  from nvidia-uvm/uvm_va_block_types.h:27,
       >                  from nvidia-uvm/uvm_hmm.h:29,
       >                  from nvidia-uvm/uvm_hmm.c:24:
       > /nix/store/jm1r8xwcalll7b85h9gippv50w7ixwzb-linux-6.19.2-dev/lib/modules/6.19.2/source/include/linux/memremap.h:227:6: note: declared here
       >   227 | void zone_device_page_init(struct page *page, struct dev_pagemap *pgmap,
       >       |      ^~~~~~~~~~~~~~~~~~~~~
       >   CC [M]  nvidia-uvm/uvm_range_tree_test.o
       > make[5]: *** [/nix/store/jm1r8xwcalll7b85h9gippv50w7ixwzb-linux-6.19.2-dev/lib/modules/6.19.2/source/scripts/Makefile.build:289: nvidia-uvm/uvm_hmm.o] Error 1
       > make[5]: *** Waiting for unfinished jobs....
       > make[4]: *** [/nix/store/jm1r8xwcalll7b85h9gippv50w7ixwzb-linux-6.19.2-dev/lib/modules/6.19.2/source/Makefile:2055: .] Error 2
       > make[3]: *** [/nix/store/jm1r8xwcalll7b85h9gippv50w7ixwzb-linux-6.19.2-dev/lib/modules/6.19.2/source/Makefile:248: __sub-make] Error 2
       > make[3]: Leaving directory '/build/source/kernel-open'
       > make[2]: *** [Makefile:248: __sub-make] Error 2
       > make[2]: Leaving directory '/nix/store/jm1r8xwcalll7b85h9gippv50w7ixwzb-linux-6.19.2-dev/lib/modules/6.19.2/source'
       > make[1]: *** [Makefile:138: modules] Error 2
       > make[1]: Leaving directory '/build/source/kernel-open'
       > make: *** [Makefile:59: modules] Error 2
       For full logs, run:
         nix log /nix/store/5skga05vis5mfv1p8mqm98hikbw1rfg3-nvidia-open-6.19.2-580.126.09.drv
error: Cannot build '/nix/store/c2y6zlv8q7zf36gjn3fn5mjz431kiq68-linux-6.19.2-modules.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/vp7zgm5x1qx93gb9bqxq0pflvhixrjzv-linux-6.19.2-modules
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.7, NixOS, 26.05 (Yarara), 26.05.20260123.c5296fd`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.3`
 - channels(root): `"nixos-25.05"`
 - nixpkgs: `/nix/store/8hx8jwa24q6rkzhca9iqy5ckk9rbphi1-source`

### Notify maintainers

@NickCao 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/nvidia-x11/default.nix
+++ b/pkgs/os-specific/linux/nvidia-x11/default.nix
@@ -61,6 +61,11 @@ let
     stripLen = 1;
     extraPrefix = "kernel/";
   };
+
+  kernel_6_19_patch = fetchpatch {
+    url = "https://github.com/CachyOS/CachyOS-PKGBUILDS/raw/d5629d64ac1f9e298c503e407225b528760ffd37/nvidia/nvidia-utils/kernel-6.19.patch";
+    hash = "sha256-YuJjSUXE6jYSuZySYGnWSNG5sfVei7vvxDcHx3K+IN4=";
+  };
 in
 rec {
   mkDriver = generic;
@@ -87,6 +92,7 @@ rec {
     openSha256 = "sha256-hECHfguzwduEfPo5pCDjWE/MjtRDhINVr4b1awFdP44=";
     settingsSha256 = "sha256-NWsqUciPa4f1ZX6f0By3yScz3pqKJV1ei9GvOF8qIEE=";
     persistencedSha256 = "sha256-wsNeuw7IaY6Qc/i/AzT/4N82lPjkwfrhxidKWUtcwW8=";
+    patchesOpen = [ kernel_6_19_patch ];
   });
 
   beta = selectHighestVersion latest (generic {

